src/Features/Core/Portable/StackTraceExplorer/ParsedStackFrame.cs

```diff
@@ -108,90 +98,49 @@ internal sealed class ParsedStackFrame : ParsedFrame
 
             return null;
 
-            // TODO: Improve perf here. ToDisplayString is fairly expensive
-            static bool MemberMatchesMethodName(ISymbol member, string memberToSearchFor)
+            static bool MatchParameters(ImmutableArray<IParameterSymbol> parameters, StackFrameParameterList stackFrameParameters)
             {
-                var displayName = member.ToDisplayString();
-                var dotIndex = displayName.LastIndexOf(".");
-                var memberName = dotIndex >= 0
-                    ? displayName[(dotIndex + 1)..]
-                    : displayName;
-
-                return string.Equals(memberName, memberToSearchFor);
-            }
-        }
-
-        /// <summary>
-        /// Gets all of the text prior to the <see cref="TypeSpan"/>
-        /// </summary>
-        /// <returns></returns>
-        public string GetTextBeforeType()
-        {
-            return OriginalText[..TypeSpan.Start];
-        }
+                if (parameters.Length != stackFrameParameters.Parameters.Length)
+                {
+                    return false;
+                }
 
-        /// <summary>
-        /// Gets the text representing the fully qualified type name
-        /// </summary>
-        public string GetQualifiedTypeText()
-        {
-            return OriginalText[TypeSpan.Start..TypeSpan.End];
-        }
+                for (var i = 0; i < stackFrameParameters.Parameters.Length; i++)
+                {
+                    var stackFrameParameter = (StackFrameParameterDeclarationNode)stackFrameParameters.Parameters[i];
+                    var paramSymbol = parameters[i];
 
-        /// <summary>
-        /// Gets the method text, including the arguments to the method
-        /// </summary>
-        public string GetMethodText()
-        {
-            return OriginalText[MethodSpan.Start..ArgsSpan.End];
-        }
+                    if (paramSymbol.Name != stackFrameParameter.Identifier.CreateString())
+                    {
+                        return false;
+                    }
+                }
 
-        /// <summary>
-        /// Gets the text after the last parsed span available. This is after
-        /// file information if it is available, otherwise after the argument information.
-        /// </summary>
-        public string GetTrailingText()
-        {
-            var lastSpan = FileSpan == default
-                ? ArgsSpan
-                : FileSpan;
+                return true;
```

Comment:
cute.  mention this won't work if it's overloads with the same name.

Reply:
Good catch. We should be able to match type here as well from the `stackFrameParameter`